packages/expo/src/winter/fetch/RequestUtils.ts

```diff
@@ -0,0 +1,151 @@
+import { ReadableStream } from 'web-streams-polyfill';
+
+import { type NativeHeadersType } from './NativeRequest';
+
+/**
+ * convert a ReadableStream to a Uint8Array
+ */
+export async function convertReadableStreamToUint8ArrayAsync(
+  stream: ReadableStream<Uint8Array>
+): Promise<Uint8Array> {
+  const reader = stream.getReader();
+  const chunks: Uint8Array[] = [];
+  let totalLength = 0;
+
+  // Read all chunks from the stream
+  while (true) {
+    const { done, value } = await reader.read();
+    if (done) break;
+    chunks.push(value);
+    totalLength += value.length;
+  }
+
+  // Concatenate all chunks into a single Uint8Array
+  const result = new Uint8Array(totalLength);
+  let offset = 0;
+  for (const chunk of chunks) {
+    result.set(chunk, offset);
+    offset += chunk.length;
+  }
+
+  return result;
+}
+
+/**
+ * Convert FormData to string
+ *
+ * `uri` is not supported for React Native's FormData.
+ * `blob` is not supported for standard FormData.
+ */
+export function convertFormData(
+  formData: FormData,
+  boundary: string = createBoundary()
+): { body: string; boundary: string } {
+  // @ts-expect-error: React Native's FormData is not compatible with the web's FormData
+  if (typeof formData.getParts !== 'function') {
+    throw new Error('Unsupported FormData implementation');
+  }
+  // @ts-expect-error: React Native's FormData is not compatible with the web's FormData
+  const parts: FormDataPart[] = formData.getParts();
+
+  const results: string[] = [];
+  for (const entry of parts) {
+    results.push(`--${boundary}\r\n`);
+    for (const [headerKey, headerValue] of Object.entries(entry.headers)) {
+      results.push(`${headerKey}: ${headerValue}\r\n`);
+    }
+    results.push(`\r\n`);
+    // @ts-expect-error: TypeScript doesn't know about the `string` property
+    if (entry.string != null) {
+      // @ts-expect-error: TypeScript doesn't know about the `string` property
+      results.push(entry.string);
+    } else {
+      throw new Error('Unsupported FormDataPart implementation');
+    }
+    results.push(`\r\n`);
+  }
+
+  results.push(`--${boundary}--\r\n`);
+  return { body: results.join(''), boundary };
+}
+
+/**
+ * Create mutipart boundary
+ */
+export function createBoundary(): string {
+  const boundaryChars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  let boundary = '----ExpoFetchFormBoundary';
```

Comment:
It's a little bit nicer to put a separator at the end. I think this is what Firefox does.
```suggestion
  let boundary = '----ExpoFetchFormBoundary-';
```

Reply:
also chromium vs firefox